File: Python/project-euler/problems/problem_016.py
from utils.code_runner import execute_code


def problem():
    current_number = 1
    number = 2
    total_power = 1000
    # Calculate the power
    for i in range(total_power):
        current_number *= number
    # Calculate the sum of the digits of the number
    digits = [int(x) for x in str(current_number)]
    total_sum = 0

    for digit in digits:
        total_sum = total_sum + digit

    return total_sum


if __name__ == '__main__':
    execute_code(problem)

File: Java/src/main/java/nl/davydehaas/projecteuler/Problem016.java
package nl.davydehaas.projecteuler;

import static nl.davydehaas.projecteuler.util.SolutionTimer.timeSolution;

import java.math.BigInteger;
import java.util.Arrays;

public class Problem016 {

    void main() {
        timeSolution(Problem016::solve);
    }

    static int solve() {
        BigInteger number = BigInteger.TWO;
        int power = 1000;

        // Calculate the power
        BigInteger currentNumber = BigInteger.ONE;
        for (int i = 0; i < power; i++) {
            currentNumber = currentNumber.multiply(number);
        }

        int[] digits = Arrays.stream(String.valueOf(currentNumber).split(""))
                .mapToInt(Integer::parseInt)
                .toArray();

        int sum = 0;
        for (int digit : digits) {
            sum += digit;
        }

        return sum;
    }
}
